internal/service/advancedclustertpf/resource.go

```diff
@@ -101,7 +101,24 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
 	if diags.HasError() {
 		return
 	}
-	cluster := r.applyClusterChanges(ctx, diags, &state, &plan)
+	stateReq := normalizeFromTFModel(ctx, &state, diags, false)
```

Comment:
refactored some applyClusterChanges logic here, to have the distinction between tenantUpgrade/ClusterChanges more clear